scene/gui/graph_edit.cpp

```diff
@@ -313,12 +313,12 @@ bool GraphEdit::is_node_connected(const StringName &p_from, int p_from_port, con
 }
 
 void GraphEdit::disconnect_node(const StringName &p_from, int p_from_port, const StringName &p_to, int p_to_port) {
-	for (const List<Ref<Connection>>::Element *E = connections.front(); E; E = E->next()) {
-		if (E->get()->from_node == p_from && E->get()->from_port == p_from_port && E->get()->to_node == p_to && E->get()->to_port == p_to_port) {
-			connection_map[p_from].erase(E->get());
-			connection_map[p_to].erase(E->get());
-			E->get()->_cache.line->queue_free();
-			connections.erase(E);
+	for (const Ref<Connection> &conn : connections) {
+		if (conn->from_node == p_from && conn->from_port == p_from_port && conn->to_node == p_to && conn->to_port == p_to_port) {
+			connection_map[p_from].erase(conn);
+			connection_map[p_to].erase(conn);
+			conn->_cache.line->queue_free();
+			connections.erase(conn);
```

Comment:
You are erasing a connection from the vector you are iterating. It seems to work correctly though, not sure how 🤔

Reply:
No, I'm iterating over `dead_connections` and erase each dead connection in `connections`.

Reply:
Am I missing something?
![image](https://github.com/user-attachments/assets/f281936b-4b9f-4197-87bf-76a6ef6fd493)
`dead_connections` don't exist in this method.

Reply:
Oh, sorry, I was looking at the wrong piece of code (which is almost identical)😅

Reply:
It's fine here since after the first matching connection is found, that connection gets erased and the function returns.
However, I was puzzled as well until I found the return statement, so rewriting the code like this makes it more readable I think:
```cpp
void GraphEdit::disconnect_node(const StringName &p_from, int p_from_port, const StringName &p_to, int p_to_port) {
	ERR_FAIL_NULL_MSG(connections_layer, "connections_layer is missing.");

	Ref<Connection> conn_to_remove;
	for (const Ref<Connection> &conn : connections) {
		if (conn->from_node == p_from && conn->from_port == p_from_port && conn->to_node == p_to && conn->to_port == p_to_port) {
			conn_to_remove = conn;
		}
	}

	if (conn_to_remove.is_valid()) {
		connection_map[p_from].erase(conn_to_remove);
		connection_map[p_to].erase(conn_to_remove);
		conn_to_remove->_cache.line->queue_free();
		connections.erase(conn_to_remove);

		minimap->queue_redraw();
		queue_redraw();
		connections_layer->queue_redraw();
		callable_mp(this, &GraphEdit::_update_top_connection_layer).call_deferred();
	}
}
```

Reply:
Ah I also missed the return. It was outside the modified code, so it didn't show.
The new code looks fine, but it's missing a `break`.

Reply:
Alright, well, this reminds me that I shouldn't code with a flu :)